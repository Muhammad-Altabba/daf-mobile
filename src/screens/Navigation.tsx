import React from 'react'
import {
  createDrawerNavigator,
  createAppContainer,
  createStackNavigator,
} from 'react-navigation'
import Settings from './Settings'
import Welcome from './Welcome'
import Logs from './Logs'
import Analytics from 'appcenter-analytics'

Analytics.setEnabled(true)

const LogsNavigator = createStackNavigator({
  LogsScreen: {
    screen: Logs,
    navigationOptions: {
      title: 'Logs',
    },
  },
})

const TabNavigator = createDrawerNavigator({
  Home: Welcome,
  Logs: LogsNavigator,
  Settings: Settings,
})

export default createAppContainer(TabNavigator)
