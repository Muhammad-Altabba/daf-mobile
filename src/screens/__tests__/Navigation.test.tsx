import 'react-native'
import React from 'react'
import Navigation from '../Navigation'
import { render } from 'react-native-testing-library'
jest.useFakeTimers()

it('renders correctly', () => {
  const tree = render(<Navigation />).toJSON()
  expect(tree).toMatchSnapshot()
})
