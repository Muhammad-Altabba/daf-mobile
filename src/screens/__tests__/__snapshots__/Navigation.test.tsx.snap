// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "overflow": "hidden",
      "zIndex": 0,
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "opacity": 1,
              },
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F5FCFF",
                "borderRadius": undefined,
                "flex": 1,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": "center",
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": undefined,
                  "borderRadius": undefined,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": 15,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#333333",
                    "fontSize": 24,
                  }
                }
              >
                Welcome to Serto
                !
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0078ff",
                  "borderRadius": 8,
                  "maxWidth": 300,
                  "padding": 15,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": undefined,
                    "borderRadius": undefined,
                    "flex": undefined,
                    "flexDirection": "row",
                    "height": undefined,
                    "justifyContent": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "width": undefined,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                    }
                  }
                >
                  change
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1000,
        }
      }
    />
  </View>
  <View
    accessibilityViewIsModal={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateX": -320,
          },
        ],
        "zIndex": 1001,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "width": 320,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <RCTScrollView
          alwaysBounceVertical={false}
        >
          <View>
            <View
              onLayout={[Function]}
              pointerEvents="box-none"
              style={
                Object {
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 20,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "paddingVertical": 4,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityLabel="Home"
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(0, 0, 0, .04)",
                        "flexDirection": "row",
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontWeight": "bold",
                            "margin": 16,
                          },
                          Object {
                            "color": "#2196f3",
                          },
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      Home
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityLabel="Logs"
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flexDirection": "row",
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontWeight": "bold",
                            "margin": 16,
                          },
                          Object {
                            "color": "rgba(0, 0, 0, .87)",
                          },
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      Logs
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityLabel="Settings"
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flexDirection": "row",
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontWeight": "bold",
                            "margin": 16,
                          },
                          Object {
                            "color": "rgba(0, 0, 0, .87)",
                          },
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      Settings
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
